scripts/exglobal_prep_sfc.sh

```diff
@@ -0,0 +1,196 @@
+# about a week.  Although not fatal, any errors must be
+# investigated.
+#
+# If there is a failure in the ice blend script AND the 6-hour old file
+# is not available as a backup, abort the script.  The global cycling
+# code can't run without an ice analysis.
+#-----------------------------------------------------------------------
+
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${BLENDED_ICE_FILE_PREV}" ]]; then
+    echo "Copy old ice blend file to current directory"
+    cpfs "${BLENDED_ICE_FILE_PREV}" "${COMOUT_OBS}/${BLENDED_ICE_FILE}"
+  else
+    err_exit "FATAL ERROR: CURRENT ICE FILE AND 6-HR OLD ICE FILE MISSING"
+  fi
+fi
+
+#-----------------------------------------------------------------------
+# now create global snow depth data for full-res gaussian grid
+# and (if a gdas run) enkf gaussian grid.
+#-----------------------------------------------------------------------
+
+export SNOW2MDLEXEC="${EXECgfs}/emcsfc_snow2mdl"
+
+LONB_CASE=$((4*${CASE:1}))
+LATB_CASE=$((2*${CASE:1}))
+
+export MODEL_SLMASK_FILE=${SLMASK:-${FIXgfs}/am/global_slmask.t${CASE:1}.${LONB_CASE}.${LATB_CASE}.grb}
+export MODEL_LATITUDE_FILE=${MDL_LATS:-${FIXgfs}/am/global_latitudes.t${CASE:1}.${LONB_CASE}.${LATB_CASE}.grb}
+export MODEL_LONGITUDE_FILE=${MDL_LONS:-${FIXgfs}/am/global_longitudes.t${CASE:1}.${LONB_CASE}.${LATB_CASE}.grb}
+export GFS_LONSPERLAT_FILE=${LONSPERLAT:-${FIXgfs}/am/global_lonsperlat.t${CASE:1}.${LONB_CASE}.${LATB_CASE}.txt}
+export MODEL_SNOW_FILE=${RUN}.${cycle}.snogrb_t${CASE:1}.${LONB_CASE}.${LATB_CASE}
+export MODEL_SNOW_FILE_PREV=${COMINobsproc_PREV}/${RUN}.${gcycle}.snogrb_t${CASE:1}.${LONB_CASE}.${LATB_CASE}
+
+echo "Create ${CASE} snow data."
+"${USHgfs}/prep_sfc_snow.sh"
+export err=$?
+
+#----------------------------------------------------------------------
+# If there was a failure in the prep_sfc_snow script, copy the 6-hr old
+# snow file to the current file.  The gfs/gdas can run with old snow data
+# for a day or two at most.   So while not fatal, any errors must be
+# investigated
+#
+# If there is a failure in the prep_sfc_snow script AND the 6-hour old
+# snow file is not available as a backup, abort the script.  The
+# global cycling can't run without an snow analysis.
+#-----------------------------------------------------------------------
+
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${MODEL_SNOW_FILE_PREV}" ]]; then
+    echo "COPY OLD ${CASE} SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
+  else
+    err_exit "CURRENT AND 6-HR OLD ${CASE} SNOW MISSING, ABORT!"
+  fi  # check of missing 6-hr snow file
+fi  # return code check
+
+#-----------------------------------------------------------------------
+# The GFS step does not run the enkf stuff.  So there is no need to
+# create an enkf snow file.
+#-----------------------------------------------------------------------
+
+if test "${RUN}" = "gfs"
+then
+  exit 0
+fi
```

Comment:
This may depend on early cycle EnKF `RUN=enkfgfs` is enabled or not.

Reply:
Updated to check if the ENKF is running for the specific RUN. Committed @ 9d6acea00.